lib/chef/provider/aws_ebs_volume.rb

```diff
@@ -68,4 +99,97 @@ def desired_options
       AWSResource.lookup_options(options, resource: new_resource)
     end
   end
+
+  def update_attachment(volume)
+    status = volume.status
+    #
+    # If we were told to attach the volume to a machine, do so
+    #
+    if desired_instance.is_a?(AWS::EC2::Instance)
+      case status
+      when :in_use
+        attachment = volume.attachments.first
+        # wrong instance
+        if attachment.instance != desired_instance
+          detach(volume, :instance => attachment.instance, :device => attachment.device)
+          attach(volume)
+        # wrong device only
+        elsif attachment.instance == desired_instance && attachment.device != new_resource.device
+          detach(volume, :device => attachment.device)
+          attach(volume)
+        end
+
+      when :available
+        attach(volume)
+      when nil
+        raise "EBS volume #{new_resource.name} does not currently exist!"
+      else
+        raise "EBS volume #{new_resource.name} (#{volume.id}) is in #{status} state!"
+      end
+
+    #
+    # If we were told to set the machine to false, detach it.
+    #
+    else
+      case status
+      when nil
+        Chef::Log.warn "EBS volume #{new_resource.name} does not currently exist!"
+      when :in_use
+        detach(volume, :instance => Chef::Resource::AwsInstance.get_aws_object(new_resource.machine, resource: new_resource))
```

Comment:
I think this is where your `option name cannot be false` error is coming from.  `new_resource.machine` is `false` in this case, so you're trying to lookup the aws object named `false` which doesn't work.  You will need to pull the machine name from the current attachments.